app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+        SPIWrite(REG_FIFO_ADDR_PTR, bytes);
+        SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+    }
+
+    private void endPacket() throws InterruptedException, IOException {
+        // put in TX mode
+        bytes.clear();
+        bytes.add((byte) (MODE_LONG_RANGE_MODE | MODE_TX));
+        SPIWrite(REG_OP_MODE, bytes);
+
+        while (true) { // Wait for TX done
+            if ((SPIRead(REG_IRQ_FLAGS, 1).get(0) & IRQ_TX_DONE_MASK) == 1) {
+                break;
+            } else {
+                Thread.sleep(100);
+            }
+        }
+        setupBytesArray(IRQ_TX_DONE_MASK);
+        SPIWrite(REG_IRQ_FLAGS, bytes);
+    }
+
+    private int parsePacket(int size) throws IOException {
+        packetLength = 0;
+        int irqFlags = SPIRead(REG_IRQ_FLAGS, 1).get(0);
+        if (size > 0) {
+            implicitHeaderMode();
+            setupBytesArray(size & 0xFF);
+            SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        } else {
+            explicitHeaderMode();
+        }
+        setupBytesArray(irqFlags);
+        SPIWrite(REG_IRQ_FLAGS, bytes);
+
+        if (((irqFlags & IRQ_RX_DONE_MASK) == 1) && ((irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 1)) {
+            packetIndex = 0;
+            if (implicitHeaderMode == 1) {
+                packetLength = SPIRead(REG_PAYLOAD_LENGTH, 1).get(0);
+            } else {
+                packetLength = SPIRead(REG_RX_NB_BYTES, 1).get(0);
+            }
+            SPIWrite(REG_FIFO_ADDR_PTR, SPIRead(REG_FIFO_RX_CURRENT_ADDR, 1));
+            idle();
+        } else if (SPIRead(REG_OP_MODE, 1).get(0) != (MODE_LONG_RANGE_MODE | MODE_RX_SINGLE)) {
+            setupBytesArray(0);
+            SPIWrite(REG_FIFO_ADDR_PTR, bytes);
+            setupBytesArray(MODE_LONG_RANGE_MODE | MODE_RX_SINGLE);
+            SPIWrite(REG_OP_MODE, bytes);
+        }
+        return packetLength;
+    }
+
+    private int packetRssi() throws IOException {
+        return SPIRead(REG_PKT_RSSI_VALUE, 1).get(0) - ((frequency < 868e6) ? 164 : 157);
+    }
+
+    private double packetSnr() throws IOException {
+        return SPIRead(REG_PKT_SNR_VALUE, 1).get(0) * 0.25;
+    }
+
+    public int write(ArrayList<Byte> byteArray) throws IOException {
+        int size = byteArray.size();
+        int currentLength = SPIRead(REG_PAYLOAD_LENGTH, 1).get(0);
+        if ((currentLength + size) > MAX_PKT_LENGTH) {
+            size = MAX_PKT_LENGTH - currentLength;
+        }
+        SPIWrite(REG_FIFO, (ArrayList<Byte>) byteArray.subList(0, size));
+        setupBytesArray(currentLength + size);
+        SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        return size;
+    }
+
+    public boolean available() throws IOException {
+        return (SPIRead(REG_RX_NB_BYTES, 1).get(0) - packetIndex) == 1;
+    }
+
+    public int checkRx() throws IOException {
+        byte irqFlags = SPIRead(REG_IRQ_FLAGS, 1).get(0);
+        if (((irqFlags & IRQ_RX_DONE_MASK) == 1) && ((irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 1)) {
```

Comment:
if (irqFlags & self.IRQ_RX_DONE_MASK) and (irqFlags & self.IRQ_PAYLOAD_CRC_ERROR_MASK)==0  in python.  which one is correct?